src/index.js

```diff
@@ -0,0 +1,277 @@
+
+        defProperty(drawGroup, layer, 'marker-fill', 'color');
+        defProperty(drawGroup, layer, 'marker-allow-overlap', 'collide');
+        defProperty(drawGroup, layer, 'marker-width', 'size');
+        defProperty(drawGroup, layer, 'marker-comp-op', 'blend');
+        defProperty(drawGroup, layer, 'marker-line-color', 'outline:color');
+        defProperty(drawGroup, layer, 'marker-line-width', 'outline:width');
+
+        delete drawGroup._hidden;
+    }
+}
+
+function processLines(yaml, layer, drawGroupName) {
+    if (layer.shader.symbolizers.indexOf('line') >= 0) {
+        yaml.filter = getFilterFn(layer, 'line');
+
+        yaml.draw[drawGroupName] = { _hidden: {} };
+        yaml.styles[drawGroupName] = { base: 'lines' };
+        const drawGroup = yaml.draw[drawGroupName];
+        //for each yaml property
+        //opacity *must* be processed first
+        defProperty(drawGroup, layer, 'line-opacity', 'opacity:general');
+
+        defProperty(drawGroup, layer, 'line-color', 'color');
+        defProperty(drawGroup, layer, 'line-width', 'width');
+        defProperty(drawGroup, layer, 'line-join', 'join');
+        defProperty(drawGroup, layer, 'line-cap', 'cap');
+        defProperty(drawGroup, layer, 'line-comp-op', 'blend');
+        defProperty(drawGroup, layer, 'line-dasharray', 'dash');
+        delete drawGroup._hidden;
+    }
+}
+
+function processPolys(yaml, layer, drawGroupName) {
+    if (layer.shader.symbolizers.indexOf('polygon') >= 0) {
+        yaml.filter = getFilterFn(layer, 'polygon');
+
+        yaml.draw[drawGroupName] = { _hidden: {} };
+        yaml.styles[drawGroupName] = { base: 'polygons' };
+        const drawGroup = yaml.draw[drawGroupName];
+        //for each yaml property
+        //opacity *must* be processed first
+        defProperty(drawGroup, layer, 'polygon-opacity', 'opacity:general');
+
+        defProperty(drawGroup, layer, 'polygon-fill', 'color');
+        defProperty(drawGroup, layer, 'polygon-comp-op', 'blend');
+
+        delete drawGroup._hidden;
+    }
+}
+
+function processStyle(yaml, drawGroupName, layerOrder) {
+    yaml.styles[drawGroupName].blend = yaml.draw[drawGroupName].blend;
+    yaml.styles[drawGroupName].blend_order = layerOrder;
+    delete yaml.draw[drawGroupName].blend;
+}
+
+function layerToYAML(layer, layerOrder) {
+    var yaml = {
+        draw: {},
+        styles: {},
+        textures: {}
+    };
+    if (layer.shader.symbolizers.length > 1) {
+        throw new Error('Multiple symbolizer on one layer is not supported');
+    } else if (layer.shader.symbolizers.length === 1) {
+        const drawGroupName = `drawGroup${layerOrder}`;
+        processPoints(yaml, layer, drawGroupName);
+        processLines(yaml, layer, drawGroupName);
+        processPolys(yaml, layer, drawGroupName);
+
+        processStyle(yaml, drawGroupName, layerOrder);
+    }
+    return yaml;
+}
+module.exports.layerToYAML = layerToYAML;
```

Comment:
I don't know if this is the best naming, as it doesn't return a YAML, but a js/json.

The same applies to all the functions (`processPoints `, ...) receiving it as a param.

Reply:
That's true, although it is the object that is used to generate the YAML, and it doesn't suffer more transformations afterwards. Do you think of a better name?

Reply:
Renamed to "scene"